File: 8-kyu-kata/filling-an-array-part-1/javascript/fillingAnArrayPart1.Test.js
// 8 kyu - Filling an array (part 1)  [ ID: 571d42206414b103dc0006a1  (filling-an-array-part-1) ]
// URL: https://www.codewars.com/kata/571d42206414b103dc0006a1
// Category: REFERENCE  |  Tags: ARRAYS | FUNDAMENTALS
// *****************************************************************************
describe("#arr creates a new array with the numbers 0 to N-1", () => {
  it("should return an array", (_) => Test.expect(arr() instanceof Array))
  it("should return a blank array when called with no argument", (_) => Test.assertSimilar(arr(), []))
  it("should return 0 to 9 when called with 10", (_) => Test.assertSimilar(arr(10), [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]))
  it("should return 0 to N-1 where N is a random number", () => {
    const _arr = (N) => [
      ...(function* () {
        let i = 0
        while (i < N) yield i++
      })()
    ]
    let N = Math.round(Math.random() * 10) + 5
    Test.assertSimilar(arr(N), _arr(N))
  })
})

File: 8-kyu-kata/filling-an-array-part-1/python/filling_an_array_part_1_test.py
# 8 kyu - Filling an array (part 1)  [ ID: 571d42206414b103dc0006a1  (filling-an-array-part-1) ]
# URL: https://www.codewars.com/kata/571d42206414b103dc0006a1
# Category: REFERENCE  |  Tags: ARRAYS | FUNDAMENTALS
# ******************************************************************************
import codewars_test as test
from solution import arr


@test.describe("Basic Tests")
def basic_tests():
    @test.it("Basic Test Cases")
    def basic_test_cases():
        test.assert_equals(arr(4), [0, 1, 2, 3])
        test.assert_equals(arr(0), [])
        test.assert_equals(arr(), [])


@test.describe("Random Tests")
def random_tests():

    from random import randint

    check = lambda n: list(range(n))

    for i in range(1, 101):
        n = randint(0, 36)
        expected = check(n)

        @test.it("Testing №{} should work for arr({})".format(i, n))
        def test_case():
            test.assert_equals(arr(n), expected)
